server/models/carModels.js

```diff
@@ -1,124 +1,121 @@
-  }
-
-  deleteCar(details) {
-    const index = this.cars.indexOf(details);
-    this.cars.splice(index, 1);
-  }
-
-  getAllCars() {
-    return this.cars;
-  }
-
-  updateOrder(details, rawData) {
-    const oldData = details;
-    this.update = {
-      id: details.id,
-      car_id: oldData.car_id,
-      status: oldData.status,
-      old_price_offered: oldData.price_offered,
-      new_price_offered: rawData.new_price_offered,
+const CarModel = {
+  async checkCarId(id) {
+    const text = {
+      name: 'fetch-car-by-id',
+      text: 'SELECT * FROM carads WHERE carid = $1',
+      values: [parseInt(id, 10)],
+    };
+    const data = await db.query(text)
+      .then(res => res).catch(e => console.error(e.stack));
+    return data;
+  },
+  async findAllAvialable() {
+    const status = 'available';
+    const text = {
+      name: 'fetch-available-cars',
+      text: 'SELECT * FROM carads WHERE status = $1',
+      values: [status],
     };
-    // update data
-    oldData.price_offered = rawData.new_price_offered;
-    return this.update;
-  }
-
-  checkOrder(req) {
-    return this.orders.find(order => order.car_id === req.params.id
-        && order.status === 'available');
-  }
-
-  checkUserOrderExisting(id) {
-    return this.orders.find(order => order.userId === id);
-  }
 
-  makeOrder(rawData, userId, details) {
-    const input = rawData;
-    input.id = this.orders.length + 1;
-    input.date_created = moment().format();
-    input.status = 'pending';
-    input.price = details.price;
-    input.userId = userId;
-    // update the list of orders
-    this.orders.push(input);
-  }
+    const data = await db.query(text)
+      .then(res => res).catch(e => console.error(e.stack));
+    return data;
+  },
+  async getCarsInPriceRange(minPrice, maxPrice) {
+    const status = 'available';
+    const text = {
 
+      name: 'fetch-available-cars-within-price-range',
+      text: 'SELECT * FROM carads WHERE carads.price BETWEEN $1 AND $2 AND status = $3;',
+      values: [minPrice, maxPrice, status],
+    };
 
-  addcarData(rawData, req) {
-    const input = rawData;
-    input.id = this.cars.length + 1;
-    input.owner = req.user.id;
-    input.date_created = moment().format();
-    // update the list of users
-    this.cars.push(input);
-  }
+    const data = await db.query(text)
+      .then(res => res).catch(e => console.error(e.stack));
+    return data;
+  },
+  async addCar(rawData, req) {
+    const text = 'INSERT INTO carads (owner, state, status, price, manufacturer, model, body_type, date_created)'
+     + 'VALUES ($1, $2, $3, $4, $5, $6, $7, $8) RETURNING *';
+    const values = [req.user.id, rawData.state, rawData.status,
+      rawData.price, rawData.manufacturer, rawData.model, rawData.body_type,
+      moment().format()];
+    const data = await db.query(text, values)
+      .then(res => res.rows[0]).catch(e => console.error(e.stack));
+    return data;
+  },
+  async updatePrice(price, id) {
```

Comment:
updatePrice and updateStatus are very similar, you can refactor to use one method that takes parameters

Reply:
alright I'll refactor that